### PR TITLE
Trigger warning if `class` is not set in `Predictor$new()` for classifcation tasks

### LINKED ISSUE #134: FeatureImp + mlr3 Learner that predicts probabilities does not work
If I want to compute the importance for a measure based on probabilities (e.g., brier score), the `FeatureImp` is never calculated on the probabilities, even if I manually use a `predict.function`: 

```
library("mlr3")
library("iml")
credit.task = tsk("german_credit")
lrn = lrn("classif.rpart", predict_type = "prob")
model = lrn$train(credit.task)
data = credit.task$data()

# write a measure that just prints the `predicted` that will be used to calculate the measure
measure_print_predicted = function(actual, predicted) {
  cat(head(predicted)) # have a look at how predicted looks like
}

pred = Predictor$new(model, data = data, y = "credit_risk")
imp = FeatureImp$new(pred, loss = measure_print_predicted, n.repetitions = 1)
# 1 2 1 2 2 1
```
It seems that internally the class is converted as numeric values (1 and 2), which makes it impossible to compute measures based on probabilities. I then tried to directly use a manually written `predict.function` which also did not work: 

```
# use a manually written predict function that returns probabilities
predict_good_prob = function(model, newdata) predict(model, newdata, predict_type = "prob")[, "good"]
head(predict_good_prob(model, data))

# use this predict function for IML method
pred = Predictor$new(model, data = data, y = "credit_risk", predict.function = predict_good_prob)
imp = FeatureImp$new(pred, loss = measure_print_predicted, n.repetitions = 1)
# 1 2 1 2 2 1
```

### PR DESCRIPTION
**Pull request by pat-s**
fixes #134 

```r
    library("mlr3")
    library("iml")
    credit.task = tsk("german_credit")
    lrn = lrn("classif.rpart", predict_type = "prob")
    model = lrn$train(credit.task)
    data = credit.task$data()

    # write a measure that just prints the `predicted` that will be used to calculate the measure
    lossf = function(actual, predicted) {
      return(predicted)
    }

    pred = Predictor$new(model, data = data, y = "credit_risk")
    #> Warning in .subset2(public_bind_env, "initialize")(...): Argument 'class' not
    #> specified but a classification problem was detected. This will likely cause
    #> issues down the road. Please supply it in `Predictor$new()`.
    FeatureImp$new(pred, loss = lossf, n.repetitions = 1)
    #> Warning in Ops.factor(permutation_error, self$original.error): '/' not
    #> meaningful for factors
    #> Error in median.default(permutation_error): need numeric data

    pred = Predictor$new(model, data = data, y = "credit_risk", class = "good")
    FeatureImp$new(pred, loss = lossf, n.repetitions = 1)
    #> Interpretation method:  FeatureImp 
    #> error function: function (actual, predicted)  {     return(predicted) }
    #> 
    #> Analysed predictor: 
    #> Prediction task: classification 
    #> Classes:  
    #> 
    #> Analysed data:
    #> Sampling from data.frame with 1000 rows and 20 columns.
    #> 
    #> Head of results:
    #>               feature importance.05 importance importance.95 permutation.error
    #> 1                 age     0.2851562          1      3.476437          0.868709
    #> 2              amount     0.1835711          1      3.506849          0.868709
    #> 3      credit_history     0.2877834          1      3.549094          0.868709
    #> 4            duration     0.2851562          1      3.474836          0.868709
    #> 5 employment_duration     0.2877834          1      3.506849          0.868709
    #> 6      foreign_worker     0.2877834          1      4.598735          0.868709

<sup>Created on 2020-07-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

### CODE REVIEW

**Comment by pat-s** [comment]
Seems to be a bad solution since the warning is not triggered for every classif task but we actually only want it for "prob" predictions.

Also it is unclear from the help pages why `class` should be set if this arg is solely for multiclass tasks.

Looking at it again.

**Comment by pat-s** [comment]
I was wrong and this PR does not change anything to the better.
